src/main/java/org/opensearch/geospatial/index/query/xyshape/XYShapeQueryProcessor.java

```diff
@@ -0,0 +1,89 @@
+import org.apache.lucene.search.MatchNoDocsQuery;
+import org.apache.lucene.search.Query;
+import org.opensearch.common.geo.ShapeRelation;
+import org.opensearch.geometry.Geometry;
+import org.opensearch.geometry.GeometryVisitor;
+import org.opensearch.geospatial.index.mapper.xyshape.XYShapeFieldMapper;
+import org.opensearch.index.mapper.MappedFieldType;
+import org.opensearch.index.query.QueryShardContext;
+import org.opensearch.index.query.QueryShardException;
+
+/**
+ * Query Processor to convert given Geometry into Lucene query
+ */
+public class XYShapeQueryProcessor {
+
+    /**
+     * Creates a {@link Query} that matches all indexed shapes to the provided {@link Geometry}  based on {@link ShapeRelation}
+     * @param geometry OpenSearch {@link Geometry} as an input
+     * @param fieldName Lucene field of type {@link XYShape}
+     * @param relation Relation to be used to get all shapes from given Geometry
+     * @param context QueryShardContext instance
+     * @return {@link Query} instance from XYShape.newGeometryQuery
+     */
+    public Query shapeQuery(
+        Geometry geometry,
+        String fieldName,
+        ShapeRelation relation,
+        GeometryVisitor<List<XYGeometry>, RuntimeException> visitor,
+        QueryShardContext context
+    ) {
+        // if no input is parsed from input, return no documents should be matched from index,
+        // this is similar to geo_shape field type
+        if (geometry == null || geometry.isEmpty()) {
+            return new MatchNoDocsQuery();
+        }
+
+        Objects.requireNonNull(fieldName, "Field name cannot be null");
+        Objects.requireNonNull(context, "QueryShardContext cannot be null");
+        validateIsXYShapeFieldType(fieldName, context);
+
+        Objects.requireNonNull(relation, "ShapeRelation cannot be null");
+        return getQueryFromGeometry(geometry, fieldName, relation.getLuceneRelation(), visitor);
+    }
+
+    private Query getQueryFromGeometry(
+        Geometry geometry,
+        String fieldName,
+        ShapeField.QueryRelation queryRelation,
+        GeometryVisitor<List<XYGeometry>, RuntimeException> visitor
+    ) {
+        final List<XYGeometry> collections = geometry.visit(visitor);
+        // if no valid geometries are found from input shpae, return No documents matched query as fallback
+        if (collections == null || collections.isEmpty()) {
+            return new MatchNoDocsQuery();
+        }
+        return XYShape.newGeometryQuery(fieldName, queryRelation, collections.toArray(new XYGeometry[0]));
+    }
+
+    private void validateIsXYShapeFieldType(String fieldName, QueryShardContext context) {
+        MappedFieldType fieldType = context.fieldMapper(fieldName);
+        if (fieldType instanceof XYShapeFieldMapper.XYShapeFieldType) {
+            return;
+        }
+        throw new QueryShardException(
+            context,
+            String.format(
+                Locale.getDefault(),
```

Comment:
I see in core OpenSearch we use ```Locale.ROOT```, not sure what is the best practice here.

Reply:
Will create an issue to investigate and will create new PR if we have to replace to Locale.ROOT. WDYT?

Reply:
Created issue here: https://github.com/opensearch-project/geospatial/issues/77